base/dcp_client.go

```diff
@@ -504,13 +504,7 @@ func (dc *DCPClient) openStreamRequest(vbID uint16) error {
 		if err == nil {
 			err = dc.verifyFailoverLog(vbID, f)
 			if err == nil {
-				e := streamOpenEvent{
-					streamEventCommon: streamEventCommon{
-						vbID: vbID,
-					},
-					failoverLogs: f,
-				}
-				dc.workerForVbno(vbID).Send(e)
+				dc.metadata.SetFailoverEntries(vbID, f)
```

Comment:
I don't feel 100% confident that the onStreamEnd usage of openStream (which calls openStreamRequest) doesn't have a scenario that would result in SetFailoverEntries could be called for a vbID while a worker is trying to process an event for the same entry while working the event queue.  

It doesn't look to me like we have any reads of the properties that are modified here(VbUUID and FailoverEntries) during worker processing, so it might be fine.  I'm still wondering whether we want a RW mutex specifically for those two properties for extra safety - what do you think?

Reply:
I feel pretty confident about this, because we call `GetMeta` right before, and this can only occur when the stream has ended, and only one `openStream` call can happen at once. I worry more about adding any additional locking here.

Reply:
I agree that this change isn't introducing any new races - thanks.